support/client/lib/vwf/utility/vertexSimplify/simplify.js

```diff
@@ -0,0 +1,188 @@
+
+                if (t > 1) {
+                    x = p2.x;
+                    y = p2.y;
+
+                } else if (t > 0) {
+                    x += dx * t;
+                    y += dy * t;
+                }
+            }
+
+            dx = p.x - x;
+            dy = p.y - y;
+
+            return dx * dx + dy * dy;
+        },
+        // rest of the code doesn't care about point format
+
+        // basic distance-based simplification
+        simplifyRadialDist: function(points, sqTolerance) {
+
+            var prevPoint = points[0],
+                newPoints = [prevPoint],
+                point;
+
+            for (var i = 1, len = points.length; i < len; i++) {
+                point = points[i];
+
+                if (this.getSqDist(point, prevPoint) > sqTolerance) {
+                    newPoints.push(point);
+                    prevPoint = point;
+                }
+            }
+
+            if (prevPoint !== point) newPoints.push(point);
+
+            return newPoints;
+        },
+
+        // simplification using optimized Douglas-Peucker algorithm with recursion elimination
+        simplifyDouglasPeucker: function(points, sqTolerance) {
+
+            var len = points.length,
+                MarkerArray = typeof Uint8Array !== 'undefined' ? Uint8Array : Array,
+                markers = new MarkerArray(len),
+                first = 0,
+                last = len - 1,
+                stack = [],
+                newPoints = [],
+                i, maxSqDist, sqDist, index;
+
+            markers[first] = markers[last] = 1;
+
+            while (last) {
+
+                maxSqDist = 0;
+
+                for (i = first + 1; i < last; i++) {
+                    sqDist = this.getSqSegDist(points[i], points[first], points[last]);
+
+                    if (sqDist > maxSqDist) {
+                        index = i;
+                        maxSqDist = sqDist;
+                    }
+                }
+
+                if (maxSqDist > sqTolerance) {
+                    markers[index] = 1;
+                    stack.push(first, index, index, last);
+                }
+
+                last = stack.pop();
+                first = stack.pop();
+            }
+
+            for (i = 0; i < len; i++) {
+                if (markers[i]) newPoints.push(points[i]);
+            }
+
+            return newPoints;
+        },
+
+        // both algorithms combined for awesome performance
+        simplify: function(points, tolerance, highestQuality) {
+
+            var sqTolerance = tolerance !== undefined ? tolerance * tolerance : 1;
+
+            points = highestQuality ? points : this.simplifyRadialDist(points, sqTolerance);
+            points = this.simplifyDouglasPeucker(points, sqTolerance);
+
+            return points;
+        },
+
+        // Simplify konva points
+        // Konva points are in an array like [x1,y1,x2,y2,...xn,yn]
+        simplifyKonvaPts: function( points ) {
```

Comment:
Did you add these functions, I assume?  What would you think about moving them out of the library code into the appropriate view drivers, such that they could be invoked from our app like `setViewProperty`, etc?

Reply:
I already had to modify the third-party code so we could load it, but I see your point.  I'll see what I can do.